nifi-nar-bundles/nifi-smb-bundle/nifi-smb-connection-pool/src/main/java/org/apache/nifi/services/smb/SmbjSessionProviderService.java

```diff
@@ -0,0 +1,175 @@
+
+@Tags({"microsoft", "samba"})
+@CapabilityDescription("Provides connection pool for ListSmb processor. ")
+public class SmbjSessionProviderService extends AbstractControllerService implements SmbSessionProviderService {
+
+    public static final PropertyDescriptor HOSTNAME = new PropertyDescriptor.Builder()
+            .displayName("Hostname")
+            .name("hostname")
+            .description("The network host of the SMB file server.")
+            .required(false)
+            .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+            .addValidator(NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor DOMAIN = new PropertyDescriptor.Builder()
+            .displayName("Domain")
+            .name("domain")
+            .description(
+                    "The domain used for authentication. Optional, in most cases username and password is sufficient.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor.Builder()
+            .displayName("Username")
+            .name("username")
+            .description(
+                    "The username used for authentication.")
+            .required(false)
+            .defaultValue("Guest")
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .displayName("Password")
+            .name("password")
+            .description("The password used for authentication.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+    public static final PropertyDescriptor PORT = new PropertyDescriptor.Builder()
+            .displayName("Port")
+            .name("port")
+            .description("Port to use for connection.")
+            .required(true)
+            .addValidator(PORT_VALIDATOR)
+            .defaultValue("445")
+            .build();
+    public static final PropertyDescriptor TIMEOUT = new PropertyDescriptor.Builder()
+            .displayName("Timeout")
+            .name("timeout")
+            .description("Timeout in seconds for read and write operations.")
+            .required(true)
+            .defaultValue("5")
+            .addValidator(INTEGER_VALIDATOR)
+            .build();
+    private static final List<PropertyDescriptor> PROPERTIES = Collections
+            .unmodifiableList(asList(
+                    HOSTNAME,
+                    DOMAIN,
+                    USERNAME,
+                    PASSWORD,
+                    PORT,
+                    TIMEOUT
+            ));
+    private SMBClient smbClient;
+    private AuthenticationContext authenticationContext;
+    private ConfigurationContext context;
+    private String hostname;
+    private Integer port;
```

Comment:
Recommend changing `port` to a primitive `int` since it is a required property.
```suggestion
    private int port;
```